internal/proto/opamp.proto

```diff
@@ -98,6 +98,9 @@ message ServerToAgent {
 
         // The server asks the agent to report addon statuses.
         ReportAddonStatus     = 0x00000002;
+
+        // The server asks the agent to restart.
+        RequestRestart        = 0x00000004;
```

Comment:
I think having this in the Flags is ambiguous. For example if both ReportEffectiveConfig and RequestRestart are set which one is the Agent supposed to do? Both? In what order?

Reply:
It could be documented that RequestRestart should take precedence. It seems unlikely that a server implementation would have a good reason to set both flags, but I agree that the behavior should be specified.

I see 3 ways to implement this capability:

1) A new message type which would have no ambiguity but require more careful handling of the unmarshal process.

```
message ServerToAgentRestartRequest {
    string instance_uid = 1;
}
```

2) A new field on the ServerToAgent message which would still be ambiguous because flags could also be set.

```
message ServerToAgent {
    string instance_uid = 1;
    ...
    bool restart_request = 
    ...
```

3) A new flag as proposed.

I am certainly open to other ideas.

Reply:
1. A new ServerToAgentRestartRequest is probably excessive since it requires changes to lots of existing spec/implementation which only expects one type of message ServerToAgent from the server to agent. I would prefer to avoid this.
2. A new field on the ServerToAgent message may be the way to go. It can be a sub-message, particularly if we expect to carry any additional data in it (do we?).
3. The ambiguity of the flag can be avoided by documenting the behavior but it is a single bit which makes passing any additional parameters impossible.

What if we make it a new field on the ServerToAgent message and call it something like ServerToAgentCommand? We can have
```protobuf
message ServerToAgentCommand {
  enum CommandType {
    RESTART,
    SHUTDOWN,
    // other commands?
  }
  CommandType type = 1;
  // maybe some other data for the command
}
```
We will need to document that either the command cannot be mixed with other ServerToAgent fields or it is applied in a particular specific order (which the spec already defines for all fields). This will remove the ambiguity.

Reply:
ServerToAgentCommand sounds like a good approach.